src/sknnr_spatial/image/_dask_backed.py

```diff
@@ -0,0 +1,139 @@
+    *,
+    estimator: BaseEstimator,
+    y=None,
+    preprocessor_cls: type[DataArrayPreprocessor] | type[DatasetPreprocessor],
+    nodata_vals=None,
+) -> DaskBackedType:
+    """Generic predict wrapper for Dask-backed arrays."""
+    check_is_fitted(estimator)
+    preprocessor = preprocessor_cls(X_image, nodata_vals=nodata_vals)
+
+    try:
+        n_targets = _infer_num_targets(y, estimator)
+        target_names = _infer_target_names(y, n_targets)
+    except TargetInferenceError:
+        msg = (
+            "The number of targets could not be inferred from the estimator. Pass a "
+            "`y` array or dataframe used to fit the estimator."
+        )
+        raise ValueError(msg) from None
+
+    y_pred = da.apply_gufunc(
+        estimator.predict,
+        "(x)->(y)",
+        preprocessor.flat,
+        axis=preprocessor.flat_band_dim,
+        output_dtypes=[float],
+        output_sizes={"y": n_targets},
+        allow_rechunk=True,
+    )
+
+    return preprocessor.unflatten(y_pred, var_names=target_names)
+
+
+def kneighbors_from_dask_backed_array(
+    X_image: DaskBackedType,
+    *,
+    estimator: KNeighborsRegressor,
+    preprocessor_cls: type[DataArrayPreprocessor] | type[DatasetPreprocessor],
+    nodata_vals=None,
+    **kneighbors_kwargs,
+) -> DaskBackedType:
+    """Generic kneighbors wrapper for Dask-backed arrays."""
+    check_is_fitted(estimator)
+    preprocessor = preprocessor_cls(X_image, nodata_vals=nodata_vals)
+    return_distance = kneighbors_kwargs.pop("return_distance", True)
+
+    k = estimator.n_neighbors
+    var_names = [f"k{i + 1}" for i in range(k)]
+
+    # Set the expected gufunc output depending on whether distances will be included
+    signature = "(x)->(k)" if not return_distance else "(x)->(k),(k)"
+    output_dtypes: list[type] = [int] if not return_distance else [float, int]
+
+    result = da.apply_gufunc(
+        estimator.kneighbors,
+        signature,
+        preprocessor.flat,
+        output_sizes={"k": k},
+        output_dtypes=output_dtypes,
+        axis=preprocessor.flat_band_dim,
+        allow_rechunk=True,
+        return_distance=return_distance,
+        **kneighbors_kwargs,
+    )
+
+    if return_distance:
+        dist, nn = result
+
+        dist = preprocessor.unflatten(dist, var_names=var_names)
+        nn = preprocessor.unflatten(nn, var_names=var_names)
+
+        return dist, nn
+
+    return preprocessor.unflatten(result, var_names=var_names)
+
+
+def _infer_target_names(y: NDArray | pd.DataFrame | None, n_targets: int) -> list[str]:
+    # Create sequential numeric target names
+    if y is None or not hasattr(y, "columns"):
+        return [f"b{i}" for i in range(n_targets)]
+
+    return y.columns
+
+
+def _infer_num_targets(y, estimator: BaseEstimator) -> int:
+    """
+    scikit-learn doesn't have a consistent standard for storing the number of targets
+    in a fitted estimator, but there are a number of common places we can look.
+    """
```

Comment:
Should this be a comment or a docstring if it's a "private" method?

Reply:
According to [PEP 8](https://peps.python.org/pep-0008/), a comment! I'll change that.

> Docstrings are not necessary for non-public methods, but you should have a comment that describes what the method does. This comment should appear after the def line.